Sources/SpeziViews/ViewModifier/ViewState/OperationState.swift

```diff
@@ -0,0 +1,63 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+
+/// The ``OperationState`` protocol is used to map the state of an operation, for example a specific action or a more complex task being performed,
+/// to a SpeziView ``ViewState``, describing the current state of a SwiftUI `View`.
+///
+/// Combined with the ``SwiftUI/View/map(state:to:)`` view modifier, the ``OperationState`` provides an easy-to-use
+/// bridging mechanism between the state of an operation and a SpeziViews ``ViewState``. One should highlight that this bridging is only done
+/// in one direction, so the data flow from the ``OperationState`` towards the ``ViewState``, not the other way around, the reason being that
+/// the state of a SwiftUI `View` doesn't influence the state of some arbitrary operation.
+///
+/// # Usage
+///
+/// ```swift
+/// public enum DownloadState {
+///     case ready
+///     case downloading(progress: Double)
+///     case error(LocalizedError)
+///     ...
+/// }
+///
+/// // Map the `DownloadState` to an `ViewState`
+/// extension DownloadState: OperationState {
+///     public var viewState: ViewState {
+///         switch self {
+///             case .ready:
+///                 .idle
+///             case .downloading:
+///                 .processing
+///             case .error(let error):
+///                 .error(error)
+///             ...
+///         }
+///     }
+/// }
+///
+/// struct StateTestView: View {
+///     @State private var downloadState: DownloadState = .ready
+///     @State private var viewState: ViewState = .idle
+///
+///     var body: some View {
+///         EmptyView()
+///             // Map the `DownloadState` to the `ViewState`
+///             .map(state: downloadState, to: $viewState)
+///             // Show alerts based on the `ViewState`
+///             .viewStateAlert(state: $viewState)
+///             .task {
+///                 // Changes to the `DownloadState`
+///                 ...
+///             }
+///     }
+/// }
+/// ```
+public protocol OperationState {
+    /// Requires the implementation of mapping logic from the ``OperationState`` to the ``ViewState``.
+    var viewState: ViewState { get }
```

Comment:
Just want to open the discussion here to maybe think of a more specific name for this property. That communicates better, that this property is used to determine the ViewState equivalence of an operation state. However, a ViewState still exists on its own and consequentially its not a bidirectional mapping. Idk, if all this can be communicated with a simple property name? My though `viewStateEquivalent`, but honestly not sure. Just to get the discussion started.

Reply:
After our in-person discussion, we settled upon 
```
var representation: ViewState { get }
```